generators/client/templates/vue/src/main/webapp/app/admin/tracker/tracker.service.ts.ejs

```diff
@@ -1,107 +1,114 @@
 import * as SockJS from 'sockjs-client';
-import * as Stomp from 'webstomp-client';
-import { Observable, Observer } from 'rxjs';
+import { Observer, map } from 'rxjs';
 import VueRouter from 'vue-router';
+import { Store } from 'vuex';
+import { RxStomp } from '@stomp/rx-stomp';
 
-export default class TrackerService {
-
-  public stompClient: Stomp.Client = null;
-  public subscriber: any = null;
-  public connection: Promise<any>;
-  public connectedPromise: any = null;
-  public listener: Observable<any>;
-  public listenerObserver: Observer<any> = null;
-  public alreadyConnectedOnce = false;
+import { AccountStateStorable } from '@/shared/config/store/account-store';
 
-  private router: VueRouter;
+const DESTINATION_TRACKER = '/topic/tracker';
+const DESTINATION_ACTIVITY = '/topic/activity';
 
-  constructor(router: VueRouter<%_ if (authenticationTypeSession) { _%>, private cookie: any<%_ } _%>) {
-    this.router = router;
-    this.connection = this.createConnection();
-    this.listener = this.createListener();
-  }
+export default class TrackerService {
+  private rxStomp: RxStomp;
 
-  public connect(): void {
-    if (this.connectedPromise === null) {
-      this.connection = this.createConnection();
-    }
-    // building absolute path so that websocket doesn't fail when deploying with a context path
-    const loc = window.location;
-    const baseHref = document.querySelector('base').getAttribute('href');
-<%_ if (authenticationTypeJwt) { _%>
-    let url = '//' + loc.host + baseHref + 'websocket/tracker';
-    const authToken = localStorage.getItem('<%=jhiPrefixDashed %>-authenticationToken') || sessionStorage.getItem('<%=jhiPrefixDashed %>-authenticationToken');
-    if (authToken) {
-      url += '?access_token=' + authToken;
-    }
-<%_ } else { _%>
-    const url = '//' + loc.host + baseHref + 'websocket/tracker';
-<%_ } _%>
-    const socket = new SockJS(url);
-    this.stompClient = Stomp.over(socket, { protocols: ['v12.stomp'] });
-    const headers = {};
+  constructor(
+      private router: VueRouter,
+      private store: Store<AccountStateStorable>,
 <%_ if (authenticationTypeSession) { _%>
-    headers['X-XSRF-TOKEN'] = this.cookie.get('JSESSIONID') || this.cookie.get('XSRF-TOKEN');
+      private cookie: any,
 <%_ } _%>
-    this.stompClient.connect(headers, () => this.afterConnect());
+    ) {
+    this.stomp = new RxStomp();
+    this.router.afterEach(() => this.sendActivity());
+
+    this.store.watch(
+      (_state, getters) => getters.authenticated,
+      (value, oldValue) => {
+        console.log('TrackerService state change', value, oldValue);
```

Comment:
we keep that?

Reply:
It's somewhat useful, not sure.
Lets drop it.
```suggestion
```